examples fail to build on recent RPM distros

# syslog-ng

## Version of syslog-ng

Today's git master snapshot.

## Platform

Recent Fedora / openSUSE / SLES releases (RHEL7, SLES12, openSUSE 42.X are not affected)

# Issue

## Failure

Compilation fails. But seeing "examples" in the name makes me wonder if we want to compile at all and if there is any way to disable compiling this...

```
[...]
  java                        : yes
  java modules                : yes
  native bindings             : auto
+ sed -i 's/libs build/libs assemble/' Makefile
+ make -j4
mkdir -p modules/java/syslog-ng-core/classes
make: *** No rule to make target 'modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.ym', needed by 'modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.y'.  Stop.
make: *** Waiting for unfinished jobs....
```

--- a/modules/examples/Makefile.am
+++ b/modules/examples/Makefile.am
@@ -1,12 +1,12 @@
 include modules/examples/sources/msg-generator/Makefile.am
 include modules/examples/sources/threaded-diskq-source/Makefile.am
+include modules/examples/sources/threaded-random-generator/Makefile.am
 
 EXAMPLE_PLUGINS = \
   $(top_builddir)/modules/examples/sources/libmsg-generator.la \
   $(top_builddir)/modules/examples/sources/libthreaded-diskq-source.la
 
 if HAVE_GETRANDOM
-include modules/examples/sources/threaded-random-generator/Makefile.am
 EXAMPLE_PLUGINS += $(top_builddir)/modules/examples/sources/libthreaded-random-generator.la
 endif
 

--- a/modules/examples/sources/threaded-random-generator/Makefile.am
+++ b/modules/examples/sources/threaded-random-generator/Makefile.am
@@ -1,3 +1,5 @@
+if HAVE_GETRANDOM
+
 noinst_LTLIBRARIES += modules/examples/sources/libthreaded-random-generator.la
 
 modules_examples_sources_libthreaded_random_generator_la_SOURCES = \
@@ -7,15 +9,6 @@ modules_examples_sources_libthreaded_random_generator_la_SOURCES = \
   modules/examples/sources/threaded-random-generator/threaded-random-generator.c \
   modules/examples/sources/threaded-random-generator/threaded-random-generator.h
 
-BUILT_SOURCES += \
-  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.y \
-  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.c \
-  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.h
-
-EXTRA_DIST += \
-  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.ym \
-  modules/examples/sources/threaded-random-generator/CMakeLists.txt
-
 modules_examples_sources_libthreaded_random_generator_la_CPPFLAGS  = \
   $(AM_CPPFLAGS) \
   -I$(top_srcdir)/modules/examples/sources/threaded-random-generator \
@@ -24,3 +17,14 @@ modules_examples_sources_libthreaded_random_generator_la_CPPFLAGS  = \
 modules_examples_sources_libthreaded_random_generator_la_LIBADD  = $(MODULE_DEPS_LIBS)
 modules_examples_sources_libthreaded_random_generator_la_LDFLAGS = $(MODULE_LDFLAGS)
 modules_examples_sources_libthreaded_random_generator_la_DEPENDENCIES = $(MODULE_DEPS_LIBS)
+
+endif
+
+BUILT_SOURCES += \
+  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.y \
+  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.c \
+  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.h
+
+EXTRA_DIST += \
+  modules/examples/sources/threaded-random-generator/threaded-random-generator-grammar.ym \
+  modules/examples/sources/threaded-random-generator/CMakeLists.txt

